src/vs/workbench/services/editor/browser/editorResolverService.ts

```diff
@@ -281,11 +290,43 @@ export class EditorResolverService extends Disposable implements IEditorResolver
 		return associations;
 	}
 
+	/**
+	 * Given the nested nature of the editors map, we should merge factories of the same glob and id to make it flat
+	 */
+	private _flattenEditorsMap() {
```

Comment:
I sugges to flatten only once and keep that in memory until you detect a change to the map. Otherwise you end up flattening for no good reason over and over.

Reply:
Haha this was actually an optimization I thought of after logging off yesterday